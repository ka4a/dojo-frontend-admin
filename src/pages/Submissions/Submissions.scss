@import "@reustleco/dojo-admin-brand/paragon/variables.scss";

.control-bar {
  display: flex;
  align-items: baseline;
  margin-top: 22px;

  &__title {
    font-size: 22px;
    margin-left: 24px;
  }
}
