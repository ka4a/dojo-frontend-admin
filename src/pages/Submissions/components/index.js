export { createPriorityCell } from './createPriorityCell';
export { createStatusCell } from './createStatusCell';
export { createSubmissionIdCell } from './createSubmissionIdCell';
