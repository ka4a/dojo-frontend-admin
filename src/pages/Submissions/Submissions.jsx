import { DataTable, DropdownFilter } from '@edx/paragon';
import React, { memo, useEffect, useMemo } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { getSubmissions } from '../../store/submissions/submissions.selector';
import { fetchSubmissions } from '../../store/submissions/submissions.thunk';
import { DEFAULT_PAGE, PAGE_SIZE } from '../../utils/constants';
import { getFilterChoicesFromColumn, sortByDate } from '../../utils';
import { BaseLayout, SubmissionsModal } from '../../components';
import { createItemLinkCell } from '../../hocs';
import { createPriorityCell, createStatusCell, createSubmissionIdCell } from './components';
import './Submissions.scss';

const INITIAL_STATE = {
  pageSize: PAGE_SIZE,
  pageIndex: DEFAULT_PAGE - 1,
};

export const Submissions = memo(() => {
  const dispatch = useDispatch();
  const { status, error, data: submissions } = useSelector(getSubmissions);

  useEffect(() => {
    dispatch(fetchSubmissions());
  }, []);

  const columns = useMemo(
    () => [
      {
        Header: 'Id',
        accessor: 'submission_id',
        disableFilters: true,
        Cell: createSubmissionIdCell,
      },
      {
        Header: 'Course',
        accessor: 'course',
        filter: 'exactText',
        filterChoices: getFilterChoicesFromColumn(submissions, 'course'),
        Cell: createItemLinkCell({
          createLink: (rowData) => rowData.course_url,
          textFieldName: 'course',
        }),
      },
      {
        Header: 'Project',
        accessor: 'unit',
        disableFilters: true,
        Cell: createItemLinkCell({
          createLink: (rowData) => rowData.unit_url,
          textFieldName: 'unit',
        }),
      },
      {
        Header: 'Student',
        accessor: 'student',
        filter: 'exactText',
        filterChoices: getFilterChoicesFromColumn(submissions, 'student'),
      },
      {
        Header: 'Priority',
        accessor: 'priority',
        disableFilters: true,
        Cell: createPriorityCell,
      },
      {
        Header: 'Status',
        accessor: 'status',
        filter: 'exactText',
        filterChoices: getFilterChoicesFromColumn(submissions, 'status'),
        Cell: createStatusCell,
      },
      {
        Header: 'Instructor',
        accessor: 'instructor_full_name',
        filter: 'exactText',
        filterChoices: getFilterChoicesFromColumn(
          submissions,
          'instructor_full_name',
        ),
        Cell: createItemLinkCell({
          createLink: (rowData) => `/coursestaff/${rowData.instructor_id}`,
          textFieldName: 'instructor_full_name',
        }),
      },
      {
        Header: 'Submission No.',
        accessor: 'submission_number',
        disableFilters: true,
      },

      {
        Header: 'Submission Time',
        accessor: 'submitedAt',
        disableFilters: true,
        Cell: ({ row }) => row.original.timeSinceSubmission,
      },
      {
        Header: 'Grading duration',
        accessor: 'gradingTimeDifference',
        disableFilters: true,
        Cell: ({ row }) => row.original.gradingTime,
      },
    ],
    [submissions],
  );

  return (
    <BaseLayout statuses={[status]} errors={[error]}>
      <div className="page-title-background d-flex align-items-center justify-content-between px-2.5">
        <div className="align-items-center">
          <div className="page-title">Submission Management</div>
        </div>
      </div>
      <DataTable
        isPaginated
        isFilterable
        isSortable
        RowStatusComponent={() => null}
        FilterStatusComponent={() => null}
        numBreakoutFilters={4}
        defaultColumnValues={{ Filter: DropdownFilter }}
        itemCount={submissions.length}
        initialState={INITIAL_STATE}
        data={sortByDate(submissions, 'submitedAt')}
        columns={columns}
      >
        <div className="control-bar">
          <p className="control-bar__title">Filter by: </p>
          <DataTable.TableControlBar />
        </div>
        <DataTable.Table />
        <DataTable.EmptyTable content="No results found" />
        <DataTable.TableFooter />
      </DataTable>
      <SubmissionsModal />
    </BaseLayout>
  );
});
