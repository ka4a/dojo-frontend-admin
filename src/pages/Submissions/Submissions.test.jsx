import { shallow, mount } from 'enzyme';
import { MemoryRouter } from 'react-router';
import { AdminDashboardProvider } from '../../test/AdminDashboardProvider';
import { Submissions } from './Submissions';

let wrapper;

describe('Submissions', () => {
  beforeEach(() => {
    wrapper = mount(
      <MemoryRouter>
        <AdminDashboardProvider>
          <Submissions />
        </AdminDashboardProvider>
      </MemoryRouter>,
    );
  });

  it('should render Submissions component', () => {
    shallow(
      <MemoryRouter>
        <AdminDashboardProvider>
          <Submissions />
        </AdminDashboardProvider>
      </MemoryRouter>,
    );
  });

  it('it should render the expected HTML', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('it should render page title', () => {
    expect(
      wrapper.contains(<div className="page-title">Submission Management</div>),
    ).toEqual(true);
  });
});
