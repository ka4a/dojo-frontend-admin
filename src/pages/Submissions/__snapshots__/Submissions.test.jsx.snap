// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Submissions it should render the expected HTML 1`] = `"<div class=\\"base-layout\\"><div class=\\"\\"><div class=\\"page-title-background d-flex align-items-center justify-content-between px-2.5\\"><div class=\\"align-items-center\\"><div class=\\"page-title\\">Submission Management</div></div></div><div class=\\"pgn__data-table-layout-wrapper\\"><div class=\\"pgn__data-table-layout-main\\"><div class=\\"pgn__data-table-wrapper\\"><div class=\\"control-bar\\"><p class=\\"control-bar__title\\">Filter by: </p><div class=\\"pgn__data-table-status-bar\\"><div class=\\"pgn__data-table-actions\\"><div class=\\"pgn__data-table-actions-left\\"><div class=\\"pgn__data-table-filters\\"><div class=\\"pgn__data-table-filters-breakout-filter\\"><div class=\\"pgn__form-group\\"><label id=\\"dropdown-filter-label-header_course-13\\" class=\\"pgn__form-label sr-only\\" for=\\"form-field14\\">Course</label><div class=\\"pgn__form-control-decorator-group\\"><select aria-labelledby=\\"dropdown-filter-label-header_course-13\\" id=\\"form-field14\\" class=\\"form-control\\"><option value=\\"\\">Course</option><option value=\\"AI Technologies\\">AI Technologies </option></select></div></div></div><div class=\\"pgn__data-table-filters-breakout-filter\\"><div class=\\"pgn__form-group\\"><label id=\\"dropdown-filter-label-header_student-15\\" class=\\"pgn__form-label sr-only\\" for=\\"form-field16\\">Student</label><div class=\\"pgn__form-control-decorator-group\\"><select aria-labelledby=\\"dropdown-filter-label-header_student-15\\" id=\\"form-field16\\" class=\\"form-control\\"><option value=\\"\\">Student</option><option value=\\"Alex Student\\">Alex Student </option></select></div></div></div><div class=\\"pgn__data-table-filters-breakout-filter\\"><div class=\\"pgn__form-group\\"><label id=\\"dropdown-filter-label-header_status-17\\" class=\\"pgn__form-label sr-only\\" for=\\"form-field18\\">Status</label><div class=\\"pgn__form-control-decorator-group\\"><select aria-labelledby=\\"dropdown-filter-label-header_status-17\\" id=\\"form-field18\\" class=\\"form-control\\"><option value=\\"\\">Status</option><option value=\\"graded\\">Graded </option></select></div></div></div><div class=\\"pgn__data-table-filters-breakout-filter\\"><div class=\\"pgn__form-group\\"><label id=\\"dropdown-filter-label-header_instructor_full_name-19\\" class=\\"pgn__form-label sr-only\\" for=\\"form-field20\\">Instructor</label><div class=\\"pgn__form-control-decorator-group\\"><select aria-labelledby=\\"dropdown-filter-label-header_instructor_full_name-19\\" id=\\"form-field20\\" class=\\"form-control\\"><option value=\\"\\">Instructor</option><option value=\\"Alexander Rassanov\\">Alexander Rassanov </option></select></div></div></div></div></div><div class=\\"pgn__data-table-actions-right\\"><div></div></div></div><div class=\\"pgn__data-table-status\\"><div class=\\"pgn__data-table-status-left\\"></div><div class=\\"pgn__data-table-toggle\\"></div></div></div></div><div class=\\"pgn__data-table-container\\"><table role=\\"table\\" class=\\"pgn__data-table is-striped\\"><thead><tr role=\\"row\\"><th colspan=\\"1\\" role=\\"columnheader\\" title=\\"Toggle SortBy\\" style=\\"cursor: pointer;\\"><span class=\\"d-flex align-items-center\\"><span>Id</span><span class=\\"pgn__icon\\" style=\\"opacity: 0.5;\\"><svg width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\" role=\\"img\\" focusable=\\"false\\" aria-hidden=\\"true\\"><path d=\\"M7 10l5-5 5 5H7zM7 14l5 5 5-5H7z\\" fill=\\"currentColor\\"></path></svg></span></span></th><th colspan=\\"1\\" role=\\"columnheader\\" title=\\"Toggle SortBy\\" style=\\"cursor: pointer;\\"><span class=\\"d-flex align-items-center\\"><span>Course</span><span class=\\"pgn__icon\\" style=\\"opacity: 0.5;\\"><svg width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\" role=\\"img\\" focusable=\\"false\\" aria-hidden=\\"true\\"><path d=\\"M7 10l5-5 5 5H7zM7 14l5 5 5-5H7z\\" fill=\\"currentColor\\"></path></svg></span></span></th><th colspan=\\"1\\" role=\\"columnheader\\" title=\\"Toggle SortBy\\" style=\\"cursor: pointer;\\"><span class=\\"d-flex align-items-center\\"><span>Project</span><span class=\\"pgn__icon\\" style=\\"opacity: 0.5;\\"><svg width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\" role=\\"img\\" focusable=\\"false\\" aria-hidden=\\"true\\"><path d=\\"M7 10l5-5 5 5H7zM7 14l5 5 5-5H7z\\" fill=\\"currentColor\\"></path></svg></span></span></th><th colspan=\\"1\\" role=\\"columnheader\\" title=\\"Toggle SortBy\\" style=\\"cursor: pointer;\\"><span class=\\"d-flex align-items-center\\"><span>Student</span><span class=\\"pgn__icon\\" style=\\"opacity: 0.5;\\"><svg width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\" role=\\"img\\" focusable=\\"false\\" aria-hidden=\\"true\\"><path d=\\"M7 10l5-5 5 5H7zM7 14l5 5 5-5H7z\\" fill=\\"currentColor\\"></path></svg></span></span></th><th colspan=\\"1\\" role=\\"columnheader\\" title=\\"Toggle SortBy\\" style=\\"cursor: pointer;\\"><span class=\\"d-flex align-items-center\\"><span>Priority</span><span class=\\"pgn__icon\\" style=\\"opacity: 0.5;\\"><svg width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\" role=\\"img\\" focusable=\\"false\\" aria-hidden=\\"true\\"><path d=\\"M7 10l5-5 5 5H7zM7 14l5 5 5-5H7z\\" fill=\\"currentColor\\"></path></svg></span></span></th><th colspan=\\"1\\" role=\\"columnheader\\" title=\\"Toggle SortBy\\" style=\\"cursor: pointer;\\"><span class=\\"d-flex align-items-center\\"><span>Status</span><span class=\\"pgn__icon\\" style=\\"opacity: 0.5;\\"><svg width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\" role=\\"img\\" focusable=\\"false\\" aria-hidden=\\"true\\"><path d=\\"M7 10l5-5 5 5H7zM7 14l5 5 5-5H7z\\" fill=\\"currentColor\\"></path></svg></span></span></th><th colspan=\\"1\\" role=\\"columnheader\\" title=\\"Toggle SortBy\\" style=\\"cursor: pointer;\\"><span class=\\"d-flex align-items-center\\"><span>Instructor</span><span class=\\"pgn__icon\\" style=\\"opacity: 0.5;\\"><svg width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\" role=\\"img\\" focusable=\\"false\\" aria-hidden=\\"true\\"><path d=\\"M7 10l5-5 5 5H7zM7 14l5 5 5-5H7z\\" fill=\\"currentColor\\"></path></svg></span></span></th><th colspan=\\"1\\" role=\\"columnheader\\" title=\\"Toggle SortBy\\" style=\\"cursor: pointer;\\"><span class=\\"d-flex align-items-center\\"><span>Submission No.</span><span class=\\"pgn__icon\\" style=\\"opacity: 0.5;\\"><svg width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\" role=\\"img\\" focusable=\\"false\\" aria-hidden=\\"true\\"><path d=\\"M7 10l5-5 5 5H7zM7 14l5 5 5-5H7z\\" fill=\\"currentColor\\"></path></svg></span></span></th><th colspan=\\"1\\" role=\\"columnheader\\" title=\\"Toggle SortBy\\" style=\\"cursor: pointer;\\"><span class=\\"d-flex align-items-center\\"><span>Submission Time</span><span class=\\"pgn__icon\\" style=\\"opacity: 0.5;\\"><svg width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\" role=\\"img\\" focusable=\\"false\\" aria-hidden=\\"true\\"><path d=\\"M7 10l5-5 5 5H7zM7 14l5 5 5-5H7z\\" fill=\\"currentColor\\"></path></svg></span></span></th><th colspan=\\"1\\" role=\\"columnheader\\" title=\\"Toggle SortBy\\" style=\\"cursor: pointer;\\"><span class=\\"d-flex align-items-center\\"><span>Grading duration</span><span class=\\"pgn__icon\\" style=\\"opacity: 0.5;\\"><svg width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\" role=\\"img\\" focusable=\\"false\\" aria-hidden=\\"true\\"><path d=\\"M7 10l5-5 5 5H7zM7 14l5 5 5-5H7z\\" fill=\\"currentColor\\"></path></svg></span></span></th></tr></thead><tbody role=\\"rowgroup\\"><tr role=\\"row\\" class=\\"pgn__data-table-row\\"><td role=\\"cell\\" class=\\"pgn__data-table-cell-wrap\\"><button class=\\"btn table-nav-link link-button\\" type=\\"button\\">1</button></td><td role=\\"cell\\" class=\\"pgn__data-table-cell-wrap\\"><a target=\\"_blank\\" rel=\\"noopener noreferrer\\" href=\\"https://learning.exp.dojoalpha.com/course/course-v1:Strata+ST101+2021Q1/home\\" class=\\"table-nav-link\\">AI Technologies</a></td><td role=\\"cell\\" class=\\"pgn__data-table-cell-wrap\\"><a target=\\"_blank\\" rel=\\"noopener noreferrer\\" href=\\"https://learning.exp.dojoalpha.com/course/course-v1:Strata+ST101+2021Q1/block-v1:Strata+ST101+2021Q1+type@sequential+block@04fe80fd1323488d9ebfe69bdf4a6ed7/block-v1:Strata+ST101+2021Q1+type@vertical+block@8b5d80b13b71413e9e1984add4be8f2d\\" class=\\"table-nav-link\\">CPP - Staging LTI</a></td><td role=\\"cell\\" class=\\"pgn__data-table-cell-wrap\\">Alex Student</td><td role=\\"cell\\" class=\\"pgn__data-table-cell-wrap\\"><span class=\\"badge badge-gray\\">Normal</span></td><td role=\\"cell\\" class=\\"pgn__data-table-cell-wrap\\"><span class=\\"badge badge-success\\">Graded</span></td><td role=\\"cell\\" class=\\"pgn__data-table-cell-wrap\\"><a target=\\"_blank\\" rel=\\"noopener noreferrer\\" href=\\"/coursestaff/70\\" class=\\"table-nav-link\\">Alexander Rassanov</a></td><td role=\\"cell\\" class=\\"pgn__data-table-cell-wrap\\">1</td><td role=\\"cell\\" class=\\"pgn__data-table-cell-wrap\\">0</td><td role=\\"cell\\" class=\\"pgn__data-table-cell-wrap\\">0</td></tr></tbody></table></div><div class=\\"pgn__data-table-footer\\"><div>Showing 1 of 1.</div><nav aria-label=\\"table pagination\\" class=\\"pagination-minimal\\"><div class=\\"sr-only\\" aria-live=\\"polite\\" aria-relevant=\\"text\\" aria-atomic=\\"true\\">Page 1, Current Page, of 1</div><ul class=\\"pagination\\"><li class=\\"page-item disabled\\"><button aria-label=\\"Previous\\" class=\\"btn-icon btn-icon-primary btn-icon-md previous page-link\\" type=\\"button\\" tabindex=\\"-1\\" disabled=\\"\\"><span class=\\"btn-icon__icon-container\\"><span class=\\"pgn__icon btn-icon__icon\\"><svg width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\" role=\\"img\\" focusable=\\"false\\" aria-hidden=\\"true\\"><path d=\\"M17.51 3.87L15.73 2.1 5.84 12l9.9 9.9 1.77-1.77L9.38 12l8.13-8.13z\\" fill=\\"currentColor\\"></path></svg></span></span></button></li><li class=\\"page-item disabled\\"><button aria-label=\\"Next\\" class=\\"btn-icon btn-icon-primary btn-icon-md next page-link\\" type=\\"button\\" tabindex=\\"-1\\" disabled=\\"\\"><span class=\\"btn-icon__icon-container\\"><span class=\\"pgn__icon btn-icon__icon\\"><svg width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\" role=\\"img\\" focusable=\\"false\\" aria-hidden=\\"true\\"><path d=\\"M6.49 20.13l1.77 1.77 9.9-9.9-9.9-9.9-1.77 1.77L14.62 12l-8.13 8.13z\\" fill=\\"currentColor\\"></path></svg></span></span></button></li></ul></nav></div></div></div></div><div></div></div></div>"`;
